lib/ansible/vars/reserved.py

```diff
@@ -70,9 +70,11 @@ def get_reserved_names(include_private=True):
 
 def warn_if_reserved(myvars):
     ''' this function warns if any variable passed conflicts with internally reserved names '''
-    reserved = get_reserved_names()
-    for varname in myvars:
-        if varname == 'vars':
-            continue  # we add this one internally
-        if varname in reserved:
-            display.warning('Found variable using reserved name: %s' % varname)
+
+    varnames = set(myvars)
+    varnames.discard('vars')  # we add this one internally, so safe to ignore
+    for varname in varnames.intersection(_RESERVED_NAMES):
+        display.warning('Found variable using reserved name: %s' % varname)
```

Comment:
currently its one per var, not sure if it makes sense per play as the var could be defined in inventory.

I'm fine making this 'less warnings'